### ISSUE TITLE
Failed to login at lightdm screen from LiveCD generated with livecd-creator

### ISSUE THREAD

**Issue by adrianharabula** [livecd-tools/livecd-tools#106]
Creating a livecd with:
```
sudo livecd-creator --verbose --config=fedora-live-cinnamon.ks --fslabel=My-Custom-Linux
```

And booting it, won't let me login further.
![image](https://user-images.githubusercontent.com/2271038/44708678-6c230300-aab0-11e8-8463-dd722ec7fc52.png)

Switching to a tty and try to login in with either root and livecd user, will refresh the screen immediately after typing username.

No error message shown.

How can I debug this further?

Tried with various fedora versions, and the result is the same:
- fedora 29
- fedora 28.1
- fedora 27.1

Repo with ks files: https://pagure.io/fedora-kickstarts
Instructions followed from: [Creating and using live CD](https://docs.fedoraproject.org/en-US/quick-docs/creating-and-using-a-live-installation-image/index.html#proc_creating-and-using-live-cd)

**Comment by adrianharabula** [comment]
Booting with `selinux=0` kernel parameter I am able at least to log in as user root in tty.

Debugging further, I see there is no cinnamon installed on livecd. After executing:
```
dnf install cinnamon
```
I am able to log in to cinnamon desktop.

**Comment by adrianharabula** [comment]
Switched to fedora workstation to avoid cinnamon session issue, now building with:
```
sudo livecd-creator --config=fedora-live-workstation.ks
```
LiveCD is able to boot only with `selinux=0` kernel parameter. I have selinux set to permissive on building machine.

I don't understand why this happens.

**Comment by Conan-Kudo** [comment]
It's most likely caused by https://github.com/livecd-tools/livecd-tools/commit/1c2307ce8b5150265f60afd7cf8a5749a1d26f9e, which was reverted in https://github.com/livecd-tools/livecd-tools/commit/230b91e4742771969e5f1820ecf40210ee00081c.

This fix should be present in the latest builds of livecd-tools on F28+.

**Comment by Conan-Kudo** [comment]
@adrianharabula I've released v26.0 as updates for Fedora 28 and 29:

* F28: https://bodhi.fedoraproject.org/updates/FEDORA-2018-a839dc3971
* F29: https://bodhi.fedoraproject.org/updates/FEDORA-2018-6d6f79359e

Could you try with this and see if it helps? If not, it may be an SELinux policy issue, which you can verify by using `enforcing=0` as a kernel boot parameter and checking the logs for AVC denials.

**Comment by notok** [comment]
Hi, @adrianharabula, @Conan-Kudo.

I recently tried to create my custom Live media, and faced same problem. (couldn't login on lightDM login screen, and also from tty.)

Tested with
```
$ rpm -qa | grep -e livecd-tools -e spin-kick
livecd-tools-26.0-3.fc29.x86_64
spin-kickstarts-0.29.2-1.fc29.noarch
```

After trying to create live media with selinux permissive mode, I found out the selinux context of the files on the media is incorrect.

audit.log; `/usr/bin/bash` is set to `root_t`, but shold be `bin_t`.
```
type=AVC msg=audit(1544146143.243:246): avc:  denied  { entrypoint } for  pid=1244 comm="login" path="/usr/bin/bash" dev="dm-0" ino=138154 scontext=unconfined_u:unconfined_r:unconfined_t:s0 tcontext=unconfined_u:object_r:root_t:s0 tclass=file permissive=1
```

I guess this is because the setfiles command [here](https://github.com/livecd-tools/livecd-tools/blob/master/imgcreate/kickstart.py#L482) is not working properly because it's not executed after chroot.



`setfiles` command used to be run after chroot like [this](https://github.com/livecd-tools/livecd-tools/commit/1c2307ce8b5150265f60afd7cf8a5749a1d26f9e#diff-29ef9eff502d344a75ab22c714c8fde7). (`self.call` executes command after chroot.) 
It's dropped in [this commit](https://github.com/livecd-tools/livecd-tools/commit/230b91e4742771969e5f1820ecf40210ee00081c).


I'll send PR to fix this.
Please check if there're no unnotified thing, or better solution; I'm new to this project 😃

**Comment by Conan-Kudo** [comment]
This should now be fixed in v26.1:

* F28: https://bodhi.fedoraproject.org/updates/FEDORA-2018-7eb5688a2f
* F29: https://bodhi.fedoraproject.org/updates/FEDORA-2018-82f85b0db9
